pytorch_lightning/accelerators/legacy/dp_accelerator.py

```diff
@@ -178,8 +178,10 @@ def reinit_scheduler_properties(self, optimizers: list, schedulers: list):
                     scheduler.load_state_dict(state)
 
     def get_reference_model(self, model) -> LightningModule:
-        if isinstance(model, LightningDataParallel):
-            return model.module
+        if isinstance(model, torch.nn.DataParallel):
+            model = model.module
+        if isinstance(model, LightningParallelModule):
+            model = model.module
```

Comment:
any difference between these two?

Reply:
yep! there is a difference. The model is wrapped like this:

DataParallel(LightingParallelModule(BoringModel))

the outermost is just the torch DP
the middle one is ours, it takes care of forwarding "forward" to the correct step method
and the one in the inner most layer is just the LightningModule from the user.

Reply:
therefore, to get to the LightningModule we potentially need to call `model.module.module`